### ISSUE TITLE
Add support for nullable reference types

### ISSUE THREAD

**Issue by qrjo** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#582]
I'm working on a project where we have nullable reference types enabled. It would be very useful to also have nullable reference types in our entities, that way we can see whether or not string values are allowed to be null, for example. Navigation properties may be harder, because you'll run into problems at initialization of a new entity.

**Comment by sjh37** [comment]
Hi @qrjo 

If you have a string which is optional, it is declared in the class as
```c#
public string ContactName { get; set; } // ContactName (length: 30)
```
The configuration class will have `.IsRequired(false)` for it. For example:
```c#
builder.Property(x => x.ContactName)
    .HasColumnName(@"ContactName")
    .HasColumnType("nvarchar(30)")
    .IsRequired(false)
    .HasMaxLength(30);
```

Check your configuration classes for your poco classes to see the `.IsRequired` settings for your properties.

**Comment by qrjo** [comment]
Yeah, that's working fine. But I want the property to be declared as
```C#
public string? ContactName { get; set; }
```

Note the question mark. That way I get warnings when I try to assign null to the property. See [here](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/nullable-reference-types). In addition to the question mark, there also has to be a `#nullable enable` directive added to the top of the file.

**Comment by sjh37** [comment]
In `EF.Reverse.POCO.v3.ttinclude` search for `public static readonly List<string> NotNullable = new List<string>`

Remove `"string",`
That will allow them to have the question marks.
I'll get back to you on `#nullable enable`

**Comment by sjh37** [comment]
Search for `public FileHeaderFooter(string subNamespace)`
Add in
```c#
header.AppendLine("#nullable enable");
```

I'll make this an option in the settings so it will be in the next release.

**Comment by sjh37** [comment]
All done. Thanks for letting me know about `#nullable enable`. This will be in the next release.

**Comment by qrjo** [comment]
It isn't quite that simple though. If you add `#nullable enable`, all reference type members need to either have a default value (like an empty string or an empty byte array) or need to be made nullable. Otherwise you're going to get a thousand warnings.

**Comment by sjh37** [comment]
Now released as part of v3.2.0
